sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -152,5 +162,36 @@ private int onConnectionEvent(final RntbdConnectionEvent event, final Throwable
 
         return 0;
     }
+
+    public void attemptBackgroundAddressRefresh(RxDocumentServiceRequest request) {
+
+        if (request.requestContext == null) {
+            return;
+        }
+
+        if (request.requestContext.isRequestCancelledOnTimeout() == null) {
+            return;
+        }
+
+        AtomicBoolean isRequestCancelledOnTimeout = request.requestContext.isRequestCancelledOnTimeout();
+        final boolean forceAddressRefresh = request.requestContext.forceRefreshAddressCache;
+
+        if (this.forceBackgroundAddressRefreshInProgress.compareAndSet(false, true)
+            && isRequestCancelledOnTimeout.get()) {
+            this.addressSelector
+                .resolveAddressesAsync(request, forceAddressRefresh)
+                .subscribeOn(Schedulers.boundedElastic())
```

Comment:
Should we use `publishOn()` as we do it for `backgroundAddressRefresh` in `RetryUtils`?

Reply:
Is there a reason for `publishOn` to be used though? `subscribeOn` will set the execution context from where the source emission occurs.